telegram_bot/telegram_bot.py

```diff
@@ -348,10 +367,43 @@ def prompt_to_repeat_group(chat_id):
     present_choices(chat_id=chat_id, choices=definitions.value_from_name("YesNo"), prompt=prompt, callback_prefix="repeat_")
 
 
+def is_valid_response(chat_id, response):
+    session = conversation_state[chat_id]
+    stage = session.stage
+    if stage.answer_type == StageType.CHOICE:
+        # lamba to get the options from the enum
+        option = list(filter(lambda x: x.val == response, definitions.value_from_name(stage.choices).options))
+        return (True if option else False)
+        
+    elif stage.answer_type == StageType.DATE:
+        # check if response is a valid date
+        from datetime import datetime
+        try:
+            datetime.strptime(response, '%Y-%m-%d')
+            return True
+        except ValueError:
+            return False
+    elif stage.answer_type == StageType.YESNO:
+        return response in [True, False]
+    else:
+        raise ValueError(f"Unknown stage type: {stage.answer_type}")
+
 def handle_user_response(chat_id, response):
     session = conversation_state[chat_id]
-    session.set_response(val=response)
+
+    # validate here the response
+    is_valid = is_valid_response(chat_id, response)
     
+    if not is_valid:
+        # print error
+        # bot.send_message(chat_id, "invalid answer, please try again")
+        # ask question again
+        ask_question(chat_id, in_new_message=True)
```

Comment:
Instead of sending the argument `in_new_message` all over the place, wouldn't it be simpler to invoke here a Session method that just deletes the last question?

Reply:
if I delete here and dont send the flag,  it will try to edit the message 
remember that the old logic just edits the old message and in this case we want to delete and add